src/browser/api/external_window.ts

```diff
@@ -693,7 +695,9 @@ function externalWindowCloseCleanup(externalWindow: Shapes.ExternalWindow): void
 
   externalWindow.emit('closed');
   externalWindow.removeAllListeners();
-  externalWindows.delete(nativeId);
+  externalWindows.delete(uuid);
+  nativeIdToUuid.delete(nativeId);
```

Comment:
We should guarantee a few of these things right, if we get an error on a preceding step we should still release the nativeUuid no?

Reply:
@rdepena Most of this cleanup seems relatively fail-proof. Wrapped the bits that might fail in a try catch.